Add Cyberwatch Galeax Parser

AmineHazi opened the pull request:
As discussed with @valentijnscholten , the default CSV and JSON reports from Cyberwatch do not include all the necessary data. Therefore, to use this parser, please utilize this tool : https://github.com/Galeax/Cyberwatch-API-DefectDojo to generate the required JSON input.

valentijnscholten commented:
Thank you @AmineHazi . To make it clear that this parser is parsing an "extended" Cyberwatch report, could you look at changing the name of the parser to something like "Cyberwatch Galeax Scan" or "Cyberwatch Scan (Galeax)" or some suggestion from your side?

valentijnscholten commented on unittests/tools/test_cyberwatch_parser.py:
```diff
@@ -0,0 +1,92 @@
+            self.assertEqual("Security Issue - Fingerprint Web Application Framework", finding.title)
+            self.assertEqual("Info", finding.severity)
+            # Expect both endpoints to have the same host as per new JSON
+            endpoint_hosts = [e.host for e in finding.unsaved_endpoints]
```
Could you add a `.clean()` for all endpoints? This validates if they are in valid format.

valentijnscholten commented on unittests/tools/test_cyberwatch_parser.py:
```diff
@@ -0,0 +1,92 @@
+
+            finding = findings[0]
+            self.assertEqual("Security Issue - Fingerprint Web Application Framework", finding.title)
+            self.assertEqual("Info", finding.severity)
```
Could you ad an `assert` on `cwe`?

valentijnscholten commented on unittests/tools/test_cyberwatch_parser.py:
Could you `assert` the actual value of the `epss` field?

AmineHazi commented:
> Thank you @AmineHazi . To make it clear that this parser is parsing an "extended" Cyberwatch report, could you look at changing the name of the parser to something like "Cyberwatch Galeax Scan" or "Cyberwatch Scan (Galeax)" or some suggestion from your side?

Thank you for your reply, @valentijnscholten. I have just changed the parser name to "Cyberwatch Scan (Galeax)". Would you like me to update the folder and file names as well, or is the new name sufficient? (For example, change the folder 'dojo\tools\cyberwatch' to 'dojo\tools\cyberwatch_galeax'.)

valentijnscholten commented:
Good idea, so yes 😀

AmineHazi commented:
> Good idea, so yes 😀

Hi @valentijnscholten, I changed the file and folder names too. I also added assertions on the CWE and EPSS fields and cleaned up the endpoints. Let me know if there's anything else to change!

valentijnscholten commented on docs/content/en/connecting_your_tools/parsers/file/cyberwatch_galeax.md:
```diff
@@ -0,0 +1,14 @@
+---
+title: "Cyberwatch"
```
```suggestion
title: "Cyberwatch (Galeax)"
```

valentijnscholten commented on docs/content/en/connecting_your_tools/parsers/file/cyberwatch_galeax.md:
```diff
@@ -0,0 +1,14 @@
+title: "Cyberwatch"
+toc_hide: true
+---
+The Cyberwatch Parser is used to import Cyberwatch scan results into DefectDojo. It accepts a JSON input that can be generated by the [Cyberwatch-API-DefectDojo](https://github.com/Galeax/Cyberwatch-API-DefectDojo) tool.
```
```suggestion
The Cyberwatch (Galeax) Parser is used to import Cyberwatch scan results into DefectDojo. It accepts a JSON input that can be generated by the [Cyberwatch-API-DefectDojo](https://github.com/Galeax/Cyberwatch-API-DefectDojo) tool created by Galeax.
```

valentijnscholten commented on docs/content/en/connecting_your_tools/parsers/file/cyberwatch_galeax.md:
```diff
@@ -0,0 +1,14 @@
+For each CVE and security issue found in the JSON input, the parser creates a corresponding finding in DefectDojo.
+
+### Sample Scan Data
+Sample Cybwerwatch Scan scans can be found [here](https://github.com/DefectDojo/django-DefectDojo/tree/master/unittests/scans/cyberwatch).
```
```suggestion
Sample Cybwerwatch Galeax Scan scans can be found [here](https://github.com/DefectDojo/django-DefectDojo/tree/master/unittests/scans/cyberwatch).
```

valentijnscholten reviewed (changes requested):
found some more references to change

valentijnscholten commented:
@AmineHazi We have scheduled this for the 2.46.0 / May 5th release. Could you look at the small final suggestions made?

AmineHazi commented:
> @AmineHazi We have scheduled this for the 2.46.0 / May 5th release. Could you look at the small final suggestions made?

Hello @valentijnscholten ! i just made the small final suggestions you made, everything should be fine now, thank you !

mtesauro reviewed (approved):
Approved
